### PR TITLE
Add a --disable-cache option to apptainer generator

### LINKED ISSUE #23305: Add a disable cache option to the apptainer generator
## Reason
There exist times when we don't want to use the apptainer cache, like when we're building controlled applications and don't want them hanging around in cache.

## Design
Add a `--disable-cache` option to `apptainer_generator.py` that will pass `--disable-cache` when necessary to all `apptainer` commands.

## Impact
Added capability.

### PR DESCRIPTION
**Pull request by loganharbour**
Closes #23305

### CODE REVIEW

File: scripts/apptainer_generator.py [hunk 1]
```diff
@@ -195,6 +197,10 @@ def apptainer_pull(self, project: str, name: str, tag: str, args=None):
         command = ['apptainer', 'pull']
         if args is not None:
             command += args
+        if (hasattr(self.args, 'disable_cache') and
+            self.args.disable_cache and
+            '--disable-cache' not in command):
```
Comment (cticenhour) [inline scripts/apptainer_generator.py:202]: One question re: this logic - where else might the attribute disable_cache be set, other than the argument flag you have added? Seems like we might not need the explicit `--disable-cache` condition here.

Comment (cticenhour) [reply scripts/apptainer_generator.py:202]: Essentially, why are we setting this up to add the flag if adding the flag sets these others?

Comment (cticenhour) [reply scripts/apptainer_generator.py:202]: Hmm....I suppose this sets up the passthrough of this flag to the other apptainer commands then.....I think I get it.

Comment (loganharbour) [reply scripts/apptainer_generator.py:202]: Are we good? 😆 can talk in person.

**Review by cticenhour — APPROVED** [review]

**Comment by loganharbour** [comment]
Sockeye failures unrelated; are due to a next break fix.

**Pull request merged**